src/mrb_curl.c

```diff
@@ -156,6 +156,24 @@ mrb_curl_perform(mrb_state *mrb, CURL* curl, mrb_value url, mrb_value headers, m
 }
 
 static mrb_value
+mrb_curl_delete(mrb_state *mrb, mrb_value self)
+{
+  CURL* curl;
+
+  mrb_value url = mrb_nil_value();
+  mrb_value headers = mrb_nil_value();
+  mrb_value b = mrb_nil_value();
+  mrb_get_args(mrb, "S|H&", &url, &headers, &b);
+
+  curl = curl_easy_init();
+  curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "DELETE");
+  curl_easy_setopt(curl, CURLOPT_UPLOAD, 0);
+  curl_easy_setopt(curl, CURLOPT_NOBODY, 0);
```

Comment:
DELETE may respond with a body: https://tools.ietf.org/html/rfc7231#section-4.3.5

`CURLOPT_NOBODY` is a confusing double-negative so `0` is correct (set to 0 to fetch DELETE response body): https://curl.haxx.se/libcurl/c/CURLOPT_NOBODY.html

Example from description of [CURLOPT_POST](https://curl.haxx.se/libcurl/c/CURLOPT_POST.html): "When setting CURLOPT_POST to 1, it will automatically set CURLOPT_NOBODY to 0."